src/MergeContext.js

```diff
@@ -570,248 +856,89 @@ class MergeContext {
-            return false;
-
-        const isFresh = await this._tagIsFresh();
-        this._log("staging tag is " + (isFresh ? "fresh" : "stale"));
-        if (isFresh) {
-            const commitStatus = await this._getStagingStatuses();
-            this._log("staging status details: " + commitStatus);
-            if (commitStatus.failed()) {
-                this._log("staging checks failed some time ago");
-                if (this._prMergeable() !== true)
-                    this._log("merge commit did not change due to conflicts with " + this._prBaseBranch());
-                return true;
-            }
-        }
-        if (!this._dryRun("deleting staging commit"))
-            await GH.deleteReference(this._stagingTag());
-        return false;
-    }
-
-    async _acquireUserProperties() {
-        const emails = await GH.getUserEmails();
-        for (let e of emails) {
-            if (e.primary) {
-                Config.githubUserEmail(e.email);
-                break;
-            }
-        }
-        assert(Config.githubUserEmail());
-
-        const user = await GH.getUser(Config.githubUserLogin());
-        Config.githubUserName(user.name);
-        assert(Config.githubUserName());
-    }
-
-    async _checkConditions() {
-        this._log("initiator: checking conditions");
-        this._approval = null;
-
-        const pr = await GH.getPR(this.prNumber(), true);
-        // refresh PR data
-        assert(pr.number === this._pr.number);
-        this._pr = pr;
-
-        if (!this._prOpen()) {
-            this._logFailedCondition("open");
-            return StepResult.Fail();
-        }
-
-        if (await this._hasLabel(Config.mergedLabel(), this.prNumber())) {
-            this._logFailedCondition("already has merged status");
-            return StepResult.Fail();
+    async _loadPrState() {
+        if (!this._tagSha) {
+            this._prState = PrState.Brewing();
+            return;
         }
 
-        const messageValid = this._prMessageValid();
-        await this._labelFailedDescription(messageValid);
-
-        this._approval = await this._checkApproval();
-        this._log("checkApproval: " + this._approval);
-        await this._setApprovalStatus(this._prHeadSha());
+        assert(this._stagedPosition);
 
-        if (this._prInProgress()) {
-            this._logFailedCondition("not in progress");
-            return StepResult.Fail();
+        if (this._stagedPosition.merged()) {
+            this._log("already merged into base some time ago");
+            this._prState = PrState.Merged();
+            return;
         }
 
-        if (!this._prMergeable()) {
-            this._logFailedCondition("mergeable");
-            return StepResult.Fail();
+        // The staged commit became out of sync with PR and(or) base branches.
+        if (!(await this._stagedCommitIsFresh())) {
+            this._log("the staged commit became stale due to PR branch and(or) base branch changes");
```

Comment:
This log message became stale. I do not think we should enumerate all _stagedCommitIsFresh() checks here:

```suggestion
            this._log("the staged commit is stale");
```

Reply:
Done.